src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -131,4 +243,18 @@ public final void checkContextAvailability(StepContext c) throws MissingContextV
     public static ExtensionList<StepDescriptor> all() {
         return ExtensionList.lookup(StepDescriptor.class);
     }
+
+    /**
+     * Convenience method to iterate all meta step descriptors.
+     */
+    public static Iterable<StepDescriptor> allMeta() {
+        return Iterables.filter(all(), new Predicate<StepDescriptor>() {
+            @Override
+            public boolean apply(@Nullable StepDescriptor i) {
+                return i!=null && i.isMetaStep();
```

Comment:
Pending `-source 8` we cannot have type parameters on the return value of `lookup`.